### PR TITLE
Count size of all journal files when vacuuming

### PR DESCRIPTION
**Pull request by dtardon**
Count size of all journal files when vacuuming. Currently, active journal files are excluded, which means that vacuuming may not remove anything even if `*MaxUse=` has been exceeded.

Also do a few assorted cleanups while at it.

### CODE REVIEW

File: src/libsystemd/sd-journal/journal-vacuum.c [hunk 1]
```diff
@@ -48,6 +48,16 @@ static int vacuum_compare(const struct vacuum_info *a, const struct vacuum_info
         return strcmp(a->filename, b->filename);
 }
 
+static void vacuum_info_array_free(vacuum_info *list, size_t n) {
+        if (!list)
+                return;
```
Comment (YHNdnzj) [inline src/libsystemd/sd-journal/journal-vacuum.c:53]: Maybe use an assert here?

```
assert(list || n == 0)
```

File: src/libsystemd/sd-journal/journal-vacuum.c [hunk 2]
```diff
@@ -48,6 +48,16 @@ static int vacuum_compare(const struct vacuum_info *a, const struct vacuum_info
...
+
+        for (vacuum_info *i = list; i < list + n; i++)
+                free(i->filename);
```
Comment (YHNdnzj) [inline src/libsystemd/sd-journal/journal-vacuum.c]: You can reuse `FOREACH_ARRAY` (not a big matter tho)

File: src/libsystemd/sd-journal/journal-vacuum.c (hunk 1 above)
Comment (dtardon) [reply src/libsystemd/sd-journal/journal-vacuum.c:53]: The coding style is to exit early if the object is `NULL`.

File: src/libsystemd/sd-journal/journal-vacuum.c (hunk 2 above)
Comment (dtardon) [reply src/libsystemd/sd-journal/journal-vacuum.c]: Indeed. I forgot about that one.

File: src/libsystemd/sd-journal/journal-vacuum.c [hunk 3]
```diff
@@ -159,6 +159,9 @@ int journal_directory_vacuum(
                 if (!S_ISREG(st.st_mode))
                         continue;
 
+                size = 512UL * (uint64_t) st.st_blocks;
+                sum += size;
```
Comment (poettering) [inline src/libsystemd/sd-journal/journal-vacuum.c]: this isn't right. we should still ignore all non-journal files, that have names that we don't recognize at all. and empty files should not be accounted either as we already delete them here, and hence definitely don't matter anymore

Comment (poettering) [reply src/libsystemd/sd-journal/journal-vacuum.c]: i.e. we should count everything that we do n_active_files++ for in the branches below.

**Pull request merged**